src/frontend/src/lib/utils/nav.utils.ts

```diff
@@ -35,8 +35,16 @@ const tokenUrl = ({
 export const networkParam = (networkId: NetworkId): string =>
 	`network=${networkId.description ?? ''}`;
 
-export const back = async ({ pop, networkId }: { pop: boolean; networkId: NetworkId }) => {
-	const rootUrl = `/?${networkParam(networkId)}`;
+export const back = async ({
+	pop,
```

Comment:
given that we add `fromUrl?: URL;` as parameter, do we still need `pop` or can we replace it by use this new parameter within the `back` function?

Reply:
right! i will change it

Reply:
Ok, since its scope is to remove the current navigation URL from the history stack, the parameter `pop` is used unrelated to the `fromRoute` two times:

```typescript
if (isRouteTransactions($page)) {
	await back({ pop: true, networkId: $networkId });
}

[...]

await back({ networkId: backToNetworkId, pop: true });
```

I would say to leave it there, what do you think? Its usage is not specifically related to the page the user was coming from.

Reply:
I see your point. After this PR, is it use somewhere else? If know, not that a strong opinion, but I would lean towards removing it anyway because for maintance reason and to scope the complexity I rather like to expose only the options that are effectively used but, we can keep it as well. I like you decide, I approve the PR.

Reply:
It is used in:

## `Network.svelte`

```typescript
	const onClick = async () => {
		await switchNetwork(network.id);

		if (isRouteTransactions($page)) {
			await back({ pop: true, networkId: $networkId });
		}

		// A small delay to give the user a visual feedback that the network is checked
		setTimeout(() => dispatch('icSelected'), 500);
	};
```

## `HideTokenModal.svelte`
```typescript
	await hideToken({
		identity: $authStore.identity
	});

	hideProgressStep = ProgressStepsHideToken.UPDATE_UI;

	// We must navigate first otherwise we might land on the default token Ethereum selected while being on network ICP.
	await back({ networkId: backToNetworkId, pop: true });

	await updateUi({
		identity: $authStore.identity
	});

	hideProgressStep = ProgressStepsHideToken.DONE;
```


But I agree that is basically always `true` so, we could change it like this

```typescript
export const back = async (params: { networkId: NetworkId; fromUrl?: URL }) => {
	const { networkId, fromUrl } = params;
	const rootUrl = fromUrl?.toString() ?? `/?${networkParam(networkId)}`;
	await goto(rootUrl, { replaceState: 'fromUrl' in params ? nonNullish(fromUrl) : true });
};
```

Reply:
Created PR #1428 for this